### DISCUSSION TITLE
Testing a loader with a parameter using @remix-run/testing

### DISCUSSION THREAD

**Discussion by mrcwbr** [remix-run/remix#6516] [Q&A] [answered]
I'm trying to test a route with a parameter. e.g. `/todos/$todoId.tsx`.

in the loader I have to access the param, but it's always undefined during the test
```ts
export async function loader({ request, params }: DataFunctionArgs) {
  const todoId = params["todoId"]; // params is always undefined in tests
  // ... load todo and return json ... 
}
```
how does this work using [@remix-run/testing](https://www.npmjs.com/package/@remix-run/testing). I already had a look at https://github.com/remix-run/remix/blob/main/packages/remix-testing/__tests__/stub-test.tsx but I couldn't find an example. 

my test currently looks like this 
```tsx
const RemixStub = unstable_createRemixStub([{
  path: '',
  element: <TodoDetail />,
  loader,
}]);

render(<RemixStub />);

// ... assert data ... 
```
can anyone help me?

**Comment by sergiodxa** [comment, marked as answer]
The reason it’s undefined is because you set the route path to `""` instead of `/:todoId` which will have the parameter.

**Reply by mrcwbr** [reply]
@sergiodxa thx for you fast response ❤️  I already tried this approach but I used the `$` symbol like in the file names. I managed to get it work with this code: 
```tsx
const todo = new Todo();
const RemixStub = unstable_createRemixStub([
  {
    path: '/todos/:todoId',
    element: <TodoDetail />,
    loader,
  },
]);

render(
  <RemixStub
    initialEntries={[`/todos/${todo.id}`]}
  />
);

expect(await screen.findByText(todo.title)).toBeDefined();
```

**Reply by kiliman** [reply]
Remember that `createRemixStub` expects a route configuration the same as React Router. So params use the RR syntax.

https://reactrouter.com/en/main/route/route
